gl.cpp

```diff
@@ -0,0 +1,164 @@
+#define SDL_MAIN_HANDLED
+#include "SDL2/SDL.h"
+#include "gl.h"
+
+// To compile: g++ fuzzy.cpp -lopengl32 -lglew32 -lSDL2 -lmingw32
+int main(int argc, char **argv)
+{  
+	SDL_Init(SDL_INIT_VIDEO);
+ 
+        SDL_GL_SetAttribute(SDL_GL_RED_SIZE, 5);
+	SDL_GL_SetAttribute(SDL_GL_GREEN_SIZE,5);
+	SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE,5);
+	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE,16);
+	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER,1);
+	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
+
+
+	SDL_Window * win = SDL_CreateWindow("OpenGL Shaders", 200, 200, SCREEN_W, SCREEN_H, SDL_WINDOW_OPENGL);
+	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
+	SDL_GL_SetSwapInterval(1);
+	SDL_GLContext cont = SDL_GL_CreateContext(win);
+	SDL_GL_MakeCurrent(win, cont);
+	glewInit();  
+	glEnable(GL_DEPTH_TEST);
+
+	bool success = true;
+	string vertexShader;
+	string fragmentShader;
+	int vertexID;
+	int fragID; 
+	int programID;
+	success &= parseFile((char*)"vertex.vs", vertexShader);
+	success &= parseFile((char*)"fragment.fs"  , fragmentShader);
+	success &= compileShader(vertexShader.c_str(), GL_VERTEX_SHADER, vertexID);
+	success &= compileShader(fragmentShader.c_str(), GL_FRAGMENT_SHADER, fragID);
+	success &= compileProgram(vertexID, fragID, programID);
+
+	/***************************************************************************************
+	* OBJECT LOADER - Vertices, UVs, Normals, etc
+	***************************************************************************************/  
+	vector<material> materials;
+	vector<vertexData> vertexBuffer;
+	bool hasUV;
+	bool hasNormal;  
+	success &= getData("pot.obj", materials, vertexBuffer, hasUV, hasNormal);
+
+	// Build out a single array of float data 
+	int stride = 3 + (2*hasUV);
+	int vertexBufferNumBytes = stride * vertexBuffer.size() * sizeof(float);
+	float* vertexBufferData = (float*)(malloc(vertexBufferNumBytes));
+	int i = 0;
+
+	// Join data together into an interleaved buffer
+	for(int vb = 0; vb < vertexBuffer.size(); vb++)
+	{
+		vertexBufferData[i++] = vertexBuffer[vb].vert[0];
+		vertexBufferData[i++] = vertexBuffer[vb].vert[1];
+		vertexBufferData[i++] = vertexBuffer[vb].vert[2];
+		if(hasUV)
+		{
+			vertexBufferData[i++] = vertexBuffer[vb].uv[0];
+			vertexBufferData[i++] = vertexBuffer[vb].uv[1];		
+		}
+		// Leave Normal data for later...
+	}
+
+	// Load in each texture 
+	vector<int> textureIDs;	
+	for(int mat = 0; mat < materials.size(); mat++)
+	{
+		int tmp;
+		material m = materials[mat];
+		success &= loadTexture(m.map_Kd, tmp);
+		textureIDs.push_back(tmp);
+	}
+	validate(success, (char*)"Setup OpenGL Program");  
```

Comment:
Where do we use the (char*)"Setup OpenGL Program" parameter? Is this the confirmation message to the terminal?